src/v3/src/components/ActionPending/ActionPending.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Box } from '@mui/material';
+import { Typography } from '@okta/odyssey-react-mui';
+import { h } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+
+import { useWidgetContext } from '../../contexts';
+import { ActionPendingElement, UISchemaElementComponent } from '../../types';
+
+/**
+ * Render component used to indicate to the user that they should wait for an action to complete.
+ * The component shows text describing the action that is pending.
+ */
+const ActionPending: UISchemaElementComponent<{
+  uischema: ActionPendingElement
+}> = (
+  { uischema: { options } },
+) => {
+  const titleRef = useRef<HTMLTitleElement>(null);
+  const { widgetProps } = useWidgetContext();
+  const { features: { autoFocus = false } = {} } = widgetProps;
+
+  useEffect(() => {
+    if (!autoFocus) {
+      titleRef.current?.focus();
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      justifyContent="center"
+      alignItems="center"
+      textAlign="center"
+      testId={options.dataSe}
+    >
+      <Typography
+        component="h2"
+        variant="h5"
```

Comment:
Should these be props/options with sensible default values? And is H2 chosen because it's the correct level after the H1 logo heading where this is most likely to appear?

Reply:
https://www.a11yproject.com/posts/how-to-accessible-heading-structure/

> Do not choose a heading by its size, but by its level in the context of the content.
> Do not skip a heading level from the top down.

I'm guessing H2 and H5 are chosen for the case(s) you focused on, but I think need flexibility for accessibility

Reply:
I created a special component for this to emphasize that this type of heading and centering is used for a specific purpose so I am using the same heading level as the `Title` component, which is also `h2` (`h1` is the logo and logo alt text)